test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
@pleerock As I understand mocha runs each test serially, without clearing the intermediate state. It makes testing difficult and error-prone, because container is filling with different services with each individual test. I think we need to implement an approach when Container is cleared or re-created before each individual test. It will make it more straightforward and robust.

However, the Container class is static right now, so we can't just re-create it. Do you have ideas how it can be solved? Do we create a `clear()` method to reset the container or do we make a container a singleton instance with ability to re-create itself?